### PR TITLE
Bump Jackson to 2.15.2

### PR DESCRIPTION
**Pull request by KaiSuchomel**
In Jackson 2.15.2 the following Issue was fixed:
https://github.com/FasterXML/jackson-databind/issues/3895

Added Test for Jackson 2.15.2 Issue.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -51,9 +51,9 @@
     </issueManagement>
 
     <properties>
-        <maven.compiler.target>11</maven.compiler.target>
-        <maven.compiler.source>11</maven.compiler.source>
-        <maven.compiler.release>11</maven.compiler.release>
+        <maven.compiler.target>17</maven.compiler.target>
+        <maven.compiler.source>17</maven.compiler.source>
+        <maven.compiler.release>17</maven.compiler.release>
```
Comment (geoand) [inline pom.xml:56]: We definitely don't want this :)

Comment (KaiSuchomel) [reply pom.xml:56]: I changed it to get my Test using "record" running. Is there another way to enable Java version with "record"??

**Comment by geoand** [comment]
Thanks for the contribution!

Please rebase the PR onto `main` (so we can get rid of the merge commit) after removing the invalid change in the `pom,xml` file I pointed out above.

File: pom.xml (hunk 1 above)
Comment (geoand) [reply pom.xml]: I would say let's remove the test altogether as it's not a Quarkus issue, it's a Jackson issue so we should not be testing for it (unless we have some problematic integration point)

Comment (KaiSuchomel) [reply pom.xml]: Is there a plan to update the maven.compiler settings to a newer version than 11? 
So far i will remove the Test.

Comment (geoand) [reply pom.xml]: > Is there a plan to update the maven.compiler settings to a newer version than 11?

No.

But for Java 17 specific things, we do have a dedicated integration test.

Comment (gsmet) [reply pom.xml]: If it’s a Jackson bug, I don’t think we need a test in our code base.

Comment (KaiSuchomel) [reply pom.xml]: Okay, than next time we need to check again manually...

Comment (geoand) [reply pom.xml]: If we were to add such tests, we would need to add them for every feature under the sun, that just isn't practical. 
What we do test extensively is our integration points with the various libraries and our workarounds for known problems. 

The Jackson problem you linked to is neither - but of course we do want to bump to the latest Jackson version.

**Comment by gsmet** [comment]
Thanks but this was incomplete. I created https://github.com/quarkusio/quarkus/pull/33846 and kept your commit and authorship in there.

**Pull request closed without merging**